### PR TITLE
Component tests for keyring behavior

### LINKED ISSUE #2231: Further test keyring unlock behavior at the UI level
We're early days doing UI-level tests, but this path is vital and deserves coverage.

> b) I'm able to unlock the wallet by entering an incorrect password - this is something we should fix in this PR - and adding a test specifically for this behavior would be excellent 👌

> This issue was related to incorrect locking of the wallet. The logic has tests already written. I have updated the wallet locking function.

_Originally posted by @0xDaedalus in https://github.com/tallycash/extension/pull/2177#pullrequestreview-1105875943_

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes #2231 

This PR adds component tests for keyring behavior at the UI level. 

Changes:
- tests for Signing Button component
- tests for Keyring Unlock component
- tests for  Shared Current Account Information component
- add test utils

Latest build: [extension-builds-2621](https://github.com/tallycash/extension/suites/9317227629/artifacts/437626806) (as of Tue, 15 Nov 2022 15:13:13 GMT).

### CODE REVIEW

File: background/main.ts [hunk 1]
```diff
@@ -234,6 +240,20 @@ const initializeStore = (preloadedState = {}, main: Main) =>
 
 type ReduxStoreType = ReturnType<typeof initializeStore>
 
+export const initializeStoreForTest = (
```
Comment (0xDaedalus) [inline background/main.ts:243]: This feels like it should be in the test-utils file

File: ui/components/AccountsNotificationPanel/__tests__/SigningButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+const getPreloadedState = (status: "locked" | "unlocked") => ({
+  keyrings: {
+    ...initialState,
+    status,
+  },
+})
+
+describe("SigningButton", () => {
+  test("should go to the unlocking page after clicking when the wallet is locked", async () => {
```
Comment (0xDaedalus) [inline ui/components/AccountsNotificationPanel/__tests__/SigningButton.test.tsx:21]: `after being clicked`

File: ui/components/AccountsNotificationPanel/__tests__/SigningButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    )
+
+    const buttonElement = ui.getByRole("button")
+
+    expect(buttonElement).toHaveTextContent(unlockText)
+    await userEvent.click(buttonElement)
+    expect(history.location.pathname).toBe("/keyring/unlock")
+  })
+
+  test("should go to the root page after clicking when the wallet is unlocked", async () => {
```
Comment (0xDaedalus) [inline ui/components/AccountsNotificationPanel/__tests__/SigningButton.test.tsx:37]: `after being clicked`

File: ui/components/Keyring/__tests__/KeyringUnlock.test.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+const getPreloadedState = (status: "locked" | "unlocked") => ({
+  keyrings: {
+    ...initialState,
+    status,
+  },
+})
+
+describe("KeyringUnlock", () => {
+  test("should not unlock the wallet", async () => {
```
Comment (0xDaedalus) [inline ui/components/Keyring/__tests__/KeyringUnlock.test.tsx:19]: We want to be extra descriptive with these test names so that in case they fail the developer looking at the failures has the maximum reasonable amount of context we can give them.  Maybe `should not unlock the wallet when an incorrect password is submitted`

File: ui/components/Keyring/__tests__/KeyringUnlock.test.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from "react"
+import userEvent from "@testing-library/user-event"
+import { initialState } from "@tallyho/tally-background/redux-slices/keyrings"
+import { renderWithProviders } from "../../../utils/test-utils"
+import KeyringUnlock from "../KeyringUnlock"
+
+const password = "password"
```
Comment (0xDaedalus) [inline ui/components/Keyring/__tests__/KeyringUnlock.test.tsx:7]:
```suggestion
const password = "an_invalid_password"
```

File: ui/components/Shared/__tests__/SharedCurrentAccountInformation.test.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import { initialState } from "@tallyho/tally-background/redux-slices/keyrings"
+import SharedCurrentAccountInformation from "../SharedCurrentAccountInformation"
+import { renderWithProviders } from "../../../utils/test-utils"
+
+const name = "Name"
+const shortenedAddress = "Shortened Address"
```
Comment (0xDaedalus) [inline ui/components/Shared/__tests__/SharedCurrentAccountInformation.test.tsx:7]:
```suggestion
const shortenedAddress = "0x208e…090cd"
```

File: ui/utils/test-utils.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React, { PropsWithChildren } from "react"
```
Comment (0xDaedalus) [inline ui/utils/test-utils.tsx:1]: What do you think about creating a `/ui/tests` folder and putting this file in there?

**Review by 0xDaedalus — CHANGES_REQUESTED** [review]
Looks great - largely non blocking comments but I do want to maintain as much separation between our test code and our application code as possible - so I think we need to move `initializeStoreForTest` out of our main application code.

File: ui/utils/test-utils.tsx (hunk 7 above)
Comment (kkosiorowska) [reply ui/utils/test-utils.tsx:1]: I think it's a great idea. Let's do it.

**Review by 0xDaedalus — APPROVED** [review]
👑
